File: output/sonarqube_analysis/codeium_solutions/java/results/q2-palindrome-number.java
/** https://leetcode.com/problems/palindrome-number */
// Problem: Palindrome Number

//Problem Description:
//Given an integer x, return true if x is a palindrome, and false otherwise.

// Constraints:
// -2^31 <= x <= 2^31 - 1

//Code Structure:


class Solution {
    public boolean isPalindrome(int x) {
        // negative numbers are not palindrome
        if (x < 0) {
            return false;
        }
        
        int reversed = 0;
        int original = x;
        while (x > 0) {
            int pop = x % 10;
            x /= 10;
            reversed = reversed * 10 + pop;
        }
        
        return original == reversed;
    }
}
    

File: output/sonarqube_analysis/copilot_solutions/python/results/q2-palindrome-number.py
# https://leetcode.com/problems/palindrome-number/

# Problem: Palindrome Number

#Problem Description:
#Given an integer x, return true if x is a palindrome, and false otherwise.

# Constraints:
# -2^31 <= x <= 2^31 - 1

#Code Structure:

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        original = x
        reversed_num = 0
        while x > 0:
            digit = x % 10
            reversed_num = reversed_num * 10 + digit
            x //= 10
        return original == reversed_num


        
    